fs/src/directory_manager.rs

```diff
@@ -210,6 +218,10 @@ pub mod tests {
                 metadata(&path).unwrap().permissions().mode() & 0o777
             );
         }
+
+        #[cfg(target_family = "windows")]
+        crate::win_permissions::tests::verify_windows_permissions(&path);
```

Comment:
I also took the opportunity to include additional checks to verify the permissions of the windows files. I relied a lot on AI to implement that part of the code. It's totally possible that there are better ways to do it, but I don't know them yet.

Reply:
Thanks.